### DISCUSSION TITLE
Github Actions for CI/CD on customer project

### DISCUSSION THREAD

**Discussion by kittiu** [Tecnativa/doodba#367] [Q&A]
Hello,

I am trying to look for way to do github actions for CI with the a project using dooda. I try to make sense with https://github.com/Tecnativa/doodba-copier-template/tree/stable/.github/workflows. Still not figure out how to apply it to our project.

I also try to make sense with doodba-qa, but don't get the concept yet, the document about it is rather limited. 
Is doodba-qa the right place to focus on for making a CI/CD? But since it is using gitlab not github, it is hard to follow. https://github.com/Tecnativa/doodba-qa/blob/master/examples/.gitlab-ci.yml

Appreciate any pointer into this topic.

Thank you !

**Comment by joao-p-marques** [comment]
Hi!

> I try to make sense with https://github.com/Tecnativa/doodba-copier-template/tree/stable/.github/workflows. Still not figure out how to apply it to our project.

Indeed, the Github workflow applied in the copier template follows a different approach to the one you need: its goal is to test the template itself, and not to set up any project, which is what you need.

Having that in consideration, doodba-qa is, indeed, the place to start. Its approach is to test and build a doodba-based project. The [provided gitlab ci example](https://github.com/Tecnativa/doodba-qa/blob/master/examples/.gitlab-ci.yml) should help you see the required jobs and stages, and have an idea of what is necessary and the tools you have available (both with the doodba base images, and the doodba-qa image itselt) that can help you in testing, although some of them are actually optional. Maybe the [Github guide on how to migrate from GitLab CI/CD to GitHub Actions](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/migrating-from-gitlab-cicd-to-github-actions) can help you. 

Fell free to ask more specific questions if this didn't help you 🙂

**Comment by kittiu** [comment]
Thank you @joao-p-marques . Yes, that was my question.

Without CI, and just on local dev machine. Do I understand correctly that we can run doodba-qa when we are inside our doodba project (i.e., cd ~/my-doodba-project).
```
kittiu@ecosoft:~/doodba/14$ docker container run --rm -it --privileged -e COMPOSE_FILE=test.yaml -v "$PWD:/project:z" -v /var/run/docker.sock:/var/run/docker.sock:z tecnativa/doodba-qa pylint
```
I got the following error though.
```
WARNING: Some networks were defined but are not used by any service: inverseproxy_shared
ERROR: Network globalwhitelist_shared declared as external, but could not be found. Please create the network manually using `docker network create globalwhitelist_shared` and try again.
```

Some arbitrary questions.

1. Why example is testing with COMPOSE_FILE=test.yaml, why not with `devel.yaml` (as we are working on devel mostly) ?
2. What is globalwhitelist ?

Thank you very much.

**Comment by yajo** [comment]
Testing a doodba project depends a lot on the infrastructure where it will be deployed and tested, and on the mental model of what and how should be tested. We don't plan to keep on maintaining doodba-qa a lot due to that, it's becoming less useful even for us. Also its usage and goals are not much friendly.

So, doodba-qa will be archived at some point, I suggest you to not rely much on it.

I suggest you to reason about what do you consider a successful test of a doodba deployment and how to test it. Maybe somebody from the community wants to start a new project targeting this usage.

**Comment by kittiu** [comment]
Thank you very much for the insight! 👍

**Comment by kittiu** [comment]
Hello, I just play around with github actions and come up with this, it works but might not be optimized. Just in case anyone interested.

```
name: test-odoo

on: [pull_request, push]

jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-python@v2
        with:
          python-version: "3.6.12"
      - uses: BSFishy/pip-action@v1
        with:
          packages: |
            docker-compose
            invoke
            pre-commit
      - run: invoke img-pull img-build --pull
      - run: invoke git-aggregate
      - run: invoke resetdb start
      - run: invoke install -e -p
```

Any improvement / suggestions welcome.
